lib/spack/spack/main.py

```diff
@@ -313,7 +309,10 @@ def add_command(self, cmd_name):
         module = spack.cmd.get_module(cmd_name)
 
         # build a list of aliases
-        alias_list = [k for k, v in aliases.items() if v == cmd_name]
+        alias_list = []
+        aliases = spack.config.get('config:aliases')
+        if aliases:
+            alias_list = [k for k, v in aliases.items() if v == cmd_name]
```

Comment:
This needs to check whether the first word of v is cmd_name, not whether they are entirely equal.

Reply:
Good catch, thanks!